test/Microsoft.Net.Http.Headers.Tests/NameValueHeaderValueTest.cs

```diff
@@ -575,6 +575,86 @@ public void TryParseStrictList_WithSomeInvlaidValues_ReturnsFalse()
             Assert.False(NameValueHeaderValue.TryParseStrictList(inputs, out results));
         }
 
+        [Theory]
+        [InlineData("value", "value")]
+        [InlineData("\"value\"", "value")]
+        [InlineData("\"hello\\\\\"", "hello\\")]
+        [InlineData("\"hello\\\"\"", "hello\"")]
+        [InlineData("\"hello\\\"foo\\\\bar\\\\baz\\\\\"", "hello\"foo\\bar\\baz\\")]
+        [InlineData("\"quoted value\"", "quoted value")]
+        [InlineData("\"quoted\\\"valuewithquote\"", "quoted\"valuewithquote")]
+        public void GetUnescapedValue_ReturnsExpectedValue(string input, string expected)
+        {
+            var header = new NameValueHeaderValue("test", input);
+
+            var actual = header.GetUnescapedValue();
+
+            Assert.Equal(expected, actual);
+        }
+
+        [Theory]
+        [InlineData("\"hello\\\"")]
+        // See https://github.com/aspnet/HttpAbstractions/issues/923
+        // Invalid escape characters in the name value do not throw a FormatException
+        // because a quoted-string isn't checked for invalid characters.
+        // Calling GetUnescapedValue() will therefore succeed.
+        public void GetUnescapedValue_ThrowsFormatException(string input)
+        {
+            var header = new NameValueHeaderValue("test", input);
+
+            Assert.Throws<FormatException>(() => header.GetUnescapedValue());
```

Comment:
This isn't good. Anything that gets past the parser should return something here, even if you have to approximate it because the parser was too lax. Only the setter should throw.

Reply:
I'm assuming we can't change the parser at this point. Should I just return "hello" here as the last escape character was invalid?

Reply:
Invalid inputs should generally be left alone rather than filtered out. E.g. return the chars `hello\`